Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -307,6 +318,7 @@ struct PathHierarchy {
                         parent.add(child: nodeWithoutSymbol, kind: nil, hash: nil)
                     }
                     parent = nodeWithoutSymbol
+                    sparseNodes[pathComponents(nodeWithoutSymbol)] = nodeWithoutSymbol
```

Comment:
I am trying to understand where sparseNodes would be used.

Reply:
The `sparseNodes` map is used above, to match existing nodes with newly-created symbol nodes:

https://github.com/swiftlang/swift-docc/pull/1212/files#diff-e0d0f88acc1bca134d47867a977e60a988a20118c4b52451f90376f6e3d56f7cR131-R133

```swift
                    if let existingSparseNode = sparseNodes.removeValue(forKey: symbol.pathComponents) {
                         node.merge(with: existingSparseNode)
                     }
```

Reply:
I see it and found it works through an example. Thank you Vera.